packages/svelte2tsx/src/nodes/slot.ts

```diff
@@ -0,0 +1,257 @@
+import { Node, walk } from 'estree-walker';
+import MagicString from 'magic-string';
+import {
+    attributeValueIsString,
+    isMember,
+    isObjectKey,
+    isObjectValueShortHand,
+    isObjectValue,
+    getSlotName,
+} from '../utils/svelteAst';
+import TemplateScope from './TemplateScope';
+import { Identifier, WithName, Let } from '../interfaces';
+
+function attributeStrValueAsJsExpression(attr: Node): string {
+    if (attr.value.length == 0) return "''"; //wut?
+
+    //handle single value
+    if (attr.value.length == 1) {
+        const attrVal = attr.value[0];
+
+        if (attrVal.type == 'Text') {
+            return '"' + attrVal.raw + '"';
+        }
+    }
+
+    // we have multiple attribute values, so we know we are building a string out of them.
+    // so return a dummy string, it will typecheck the same :)
+    return '"__svelte_ts_string"';
+}
+
+export class SlotHandler {
+
+    constructor(private readonly str: MagicString, private readonly htmlx: string) { }
+
+    slots = new Map<string, Map<string, string>>();
+    resolved = new Map<WithName, string>();
+    resolvedExpression = new Map<Node, string>();
+
+    resolve(identifierDef: Identifier, initExpression: Node, scope: TemplateScope) {
+        let resolved = this.resolved.get(identifierDef);
+        if (resolved) {
+            return resolved;
+        }
+
+        resolved = this.getResolveExpressionStr(identifierDef, scope, initExpression);
+
+        this.resolved.set(identifierDef, resolved);
+
+        return resolved;
+    }
+
+    private getResolveExpressionStr(
+        identifierDef: Identifier,
+        scope: TemplateScope,
+        initExpression: Node
+    ) {
+        const { name } = identifierDef;
+
+        const owner = scope.getOwner(name);
```

Comment:
`owner` is possibly undefined, another check is needed for this.

(note to self: we need to turn on `strictMode` someday for `svelte2tsx`)